### PR TITLE
Produce human-readable suggestions.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: test/end2end/pub_server_data.dart [hunk 1]
```diff
@@ -304,7 +304,17 @@ final _data = {
...
+    'warningCount': 1,
+    'hintCount': 0,
   },
   'fitness': {'magnitude': 455.0, 'shortcoming': 1.0},
+  'suggestions': [
+    {
+      'level': 'warning',
+      'title': 'Use `analysis_options.yaml`.',
+      'description':
+          'Static analysis will help you to detect bugs and potential issues earlier.',
```
Comment (kevmoo) [inline test/end2end/pub_server_data.dart]: We're analyzing (in pana) all the time. The important bit is that the user has strong-mode enabled.

That really should be the check here...

Comment (isoos) [reply test/end2end/pub_server_data.dart]: PTAL, added strong-mode check.

File: lib/src/maintenance.dart [hunk 2]
```diff
@@ -154,14 +173,64 @@ Future<Maintenance> detectMaintenance(
...
+  if (analysisOptionsExists) {
+    for (var name in analysisOptionsFiles) {
+      final file = new File(p.join(pkgDir, name));
+      if (await file.exists()) {
+        final content = await file.readAsString();
+        try {
+          final Map map = yaml.loadYaml(content);
+          final analyzer = map['analyzer'];
+          if (analyzer != null) {
+            strongModeEnabled = analyzer['strong-mode'] == true;
```
Comment (kevmoo) [inline lib/src/maintenance.dart]: Or! `analyzer['strong-mode']` is a Map – see https://www.dartlang.org/guides/language/analysis-options#specifying-strong-mode

Comment (isoos) [reply lib/src/maintenance.dart]: Good catch! Fixed...

**Pull request merged**